integrationtests/smb_test.go

```diff
@@ -0,0 +1,195 @@
+
+var seededRand *rand.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func stringWithCharset(length int, charset string) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+// RandomString generates a random string with specified length
+func randomString(length int) string {
+	return stringWithCharset(length, letterset)
+}
+
+func setupUser(username, password string) error {
+	cmdLine := fmt.Sprintf(`$PWord = ConvertTo-SecureString $Env:password -AsPlainText -Force` +
+		`;New-Localuser -name $Env:username -accountneverexpires -password $PWord`)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("username=%s", username),
+		fmt.Sprintf("password=%s", password))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("setupUser failed: %v, output: %q", err, string(output))
+	}
+	return nil
+}
+
+func removeUser(username string) error {
+	cmdLine := fmt.Sprintf(`Remove-Localuser -name $Env:username`)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("username=%s", username))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("setupUser failed: %v, output: %q", err, string(output))
+	}
+	return nil
+}
+
+func setupSmbShare(shareName, localPath, username string) error {
+	if err := os.MkdirAll(localPath, 0755); err != nil {
+		return fmt.Errorf("setupSmbShare failed to create local path %q: %v", localPath, err)
+	}
+	cmdLine := fmt.Sprintf(`New-SMBShare -Name $Env:sharename -Path $Env:path -fullaccess $Env:username`)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("sharename=%s", shareName),
+		fmt.Sprintf("path=%s", localPath),
+		fmt.Sprintf("username=%s", username))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("setupSmbShare failed: %v, output: %q", err, string(output))
+	}
+
+	return nil
+}
+
+func removeSmbShare(shareName string) error {
+	cmdLine := fmt.Sprintf(`Remove-SMBShare -Name $Env:sharename -Force`)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("sharename=%s", shareName))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("setupSmbShare failed: %v, output: %q", err, string(output))
+	}
+	return nil
+}
+
+func getSmbGlobalMapping(remotePath string) error {
+	// use PowerShell Environment Variables to store user input string to prevent command line injection
+	// https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-5.1
+	cmdLine := fmt.Sprintf(`Get-SmbGlobalMapping -RemotePath $Env:smbremotepath`)
+
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("smbremotepath=%s", remotePath))
+	output, err := cmd.CombinedOutput()
```

Comment:
Does this `output` have any data that can be used in the caller of `getSmbGlobalMapping` i.e. in the test at https://github.com/kubernetes-csi/csi-proxy/pull/23/files#diff-67929a78f59e5017ffd3d88459a0098bR188 to do a more rigorous verification (besides whether err != nil)?

Reply:
it could return status, so I check whether the status is OK or not for verification.